parlai/core/torch_generator_agent.py

```diff
@@ -894,7 +894,8 @@ def _generate(self, batch, beam_size, max_ts):
             score = score.view(bsz, beam_size, -1)
             if self.temperature != 1.0:
                 score.div_(self.temperature)
-            score = F.log_softmax(score, dim=-1)
+            # force to fp32 to avoid overflow issues during search calculations
+            score = F.log_softmax(score, dim=-1, dtype=torch.float32)
             for i, b in enumerate(beams):
                 if not b.is_done():
                     b.advance(score[i])
```

Comment:
@stephenroller In the case that self.fp16 is true, should we convert back to `dtype=torch.float16` and clamp to neginf after the search to avoid downstream memory issues?

Reply:
https://github.com/facebookresearch/ParlAI/blob/26edaec671342d4a3a092513e18601bca20a8708/parlai/core/torch_generator_agent.py#L435 We do the loss in fp32. Let's just keep it fp32 and see if it breaks. The scores are moved to CPU anyway.